linters/importorder/main.go

```diff
@@ -0,0 +1,373 @@
+			if goldImport != originalGroups[i].Imports[j] {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func getGoldStandard(imports []importDecl, patterns []string) (importDecl, error) {
+	var emptyImportDecl importDecl
+	if len(imports) > 1 {
+		return emptyImportDecl, errMultipleImport
+	}
+
+	if len(imports) == 0 {
+		return emptyImportDecl, nil
+	}
+
+	combinedImports := concatenateImports(imports[0])
+	goldStandard := createGoldStandard(combinedImports, patterns)
+
+	return goldStandard, nil
+}
+
+func createGoldStandard(imports []importSpec, patterns []string) importDecl {
+	groups := make([]importGroup, 0, len(patterns))
+	importMap := convertToMap(imports)
+
+	for i, pattern := range patterns {
+		var tempGroup []importSpec
+		for imp := range importMap {
+			if pattern == standardImportGroup {
+				tempGroup = orderStdLib(tempGroup, imp, importMap)
+			}
+			if pattern == externalImportGroup {
+				tempGroup = orderExt(tempGroup, imp, importMap, patterns, i)
+			}
+			if strings.Contains(imp.Path, pattern) {
+				tempGroup = append(tempGroup, imp)
+				delete(importMap, imp)
+			}
+		}
+		sort.Sort(importSpecs(tempGroup))
+		if len(tempGroup) > 0 {
+			importGroup := importGroup{Imports: tempGroup}
+			groups = append(groups, importGroup)
+		}
+	}
+	return importDecl{
+		Groups: groups,
+	}
+}
+
+func orderExt(tempGroup []importSpec, imp importSpec, importMap map[importSpec]interface{}, patterns []string, i int) []importSpec {
+	if i == len(patterns)-1 {
+		if strings.Contains(imp.Path, ".") {
+			tempGroup = append(tempGroup, imp)
+			delete(importMap, imp)
+		}
+	} else {
+		for _, pattern := range patterns[i+1:] {
+			if !strings.Contains(imp.Path, pattern) {
+				tempGroup = append(tempGroup, imp)
+				delete(importMap, imp)
+			}
+		}
+	}
+	return tempGroup
+}
+
+func orderStdLib(tempGroup []importSpec, imp importSpec, importMap map[importSpec]interface{}) []importSpec {
+	if !isThirdParty(imp.Path) {
+		tempGroup = append(tempGroup, imp)
+		delete(importMap, imp)
+	}
+	return tempGroup
+}
+
+func convertToMap(imports []importSpec) map[importSpec]interface{} {
+	importsMap := make(map[importSpec]interface{})
+	for _, imp := range imports {
+		importsMap[imp] = struct{}{}
+	}
+	return importsMap
+}
+
+func concatenateImports(imports importDecl) []importSpec {
+	var combinedImports []importSpec
+	for _, group := range imports.Groups {
+		for _, imp := range group.Imports {
+			combinedImports = append(combinedImports, imp)
+		}
+	}
+	return combinedImports
+}
+
+// importDecl is the collection of importGroups contained in a single import block.
+type importDecl struct {
+	Groups []importGroup
+}
+
+// importGroup is a collection of imports
+type importGroup struct {
+	Imports importSpecs
+}
+
+type importSpecs []importSpec
+
+func (imp importSpecs) Len() int {
+	return len(imp)
+}
+
+func (imp importSpecs) Swap(i, j int) {
+	imp[i], imp[j] = imp[j], imp[i]
+}
+
+func (imp importSpecs) Less(i, j int) bool {
```

Comment:
could you validate that this sorts the same ways as `gofmt` and `goimports`

Reply:
Looks like [it does](https://github.com/golang/go/blob/bc8bb5b27a5abe48fbfc79ae4a1ce4c87ca90bbc/src/go/ast/import.go#L141).

Reply:
Cool, sounds good.